framework/include/base/MooseApp.h

```diff
@@ -1115,6 +1132,20 @@ class MooseApp : public ConsoleStreamInterface,
   std::map<std::pair<std::string, std::string>, void *> _lib_handles;
 
 private:
+  ///@{
+  /// Structs that are used in the _interface_registry
+  struct InterfaceRegistryObjectsBase
+  {
+    ~InterfaceRegistryObjectsBase() {}
+  };
+
+  template <class T>
+  struct InterfaceRegistryObjects : public InterfaceRegistryObjectsBase
+  {
+    std::vector<T *> _objects;
```

Comment:
otherwise this vector does not get destroyed properly (and hence you leak memory when pushing into it)